src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,59 @@
+import { useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { getSearchWith } from '../utils/searchHelper';
+import { SearchLink } from './SearchLink';
+import classNames from 'classnames';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const centuries = searchParams.getAll('centuries' || []);
+  const [input, setInput] = useState('');
+
+  function handleFilterInput(event: React.ChangeEvent<HTMLInputElement>) {
+    const search = getSearchWith(searchParams, {
+      query: event.target.value || null,
+    });
+
+    setInput(event.target.value);
+
+    setSearchParams(search);
+  }
+
+  function toggleCenturies(ch: string): string[] {
+    return centuries.includes(ch)
+      ? centuries.filter(century => century !== ch)
+      : [...centuries, ch];
+  }
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">
+        <SearchLink
+          params={{ sex: null }}
+          className={classNames({
+            'is-active': searchParams.get('sex') === null,
+          })}
+        >
           All
-        </a>
-        <a className="" href="#/people?sex=m">
+        </SearchLink>
+        <SearchLink
+          params={{ sex: 'm' }}
+          className={classNames({
+            'is-active': searchParams.get('sex') === 'm',
```

Comment:
```suggestion
            'is-active': searchParams.get('sex') === MAN,
```
Consider creating a variable